Realm settings(Email): Adds email tab to realm settings section

jenny-s51 opened the pull request:
## Motivation
<!-- Add references to relevant tickets, issues, design specs and/or a short description of what motivated you to do it. -->
https://issues.redhat.com/browse/APPDUX-935

## Verification Steps
<!--
Add the steps required to verify this change. Keep in mind that these steps should be written so groups unfamiliar with the 
new functionality can follow them, such as QE or documentation.-->

1. Go to Realm settings -> Email tab
2. Populate fields and test checkboxes/switches
3. Click save and navigate away from the page.
4. Return to the Email tab and verify the data is stored/saved correctly.

## Checklist:

- [x] Code has been tested locally by PR requester
- [x] User-visible strings are using the react-i18next framework (useTranslation)
- [ ] Help has been implemented
- [x] axe report has been run and resulting a11y issues have been resolved
- [x] Unit tests have been created/updated
- [x] Formatting has been performed via prettier/eslint
- [x] Type checking has been performed via 'yarn check-types'

ssilvert commented:
@jenny-s51 Please fix conflicts.

jonkoops commented on cypress/support/pages/admin_console/manage/realm_settings/RealmSettingsPage.ts:
```diff
@@ -1,5 +1,6 @@
 export default class RealmSettingsPage {
-    saveBtn: string;
+    generalSaveBtn: string;
```
These field declarations can be initialized directly so they do not have to be assigned in the constructor. For example:

```suggestion
    generalSaveBtn = "general-tab-save"
```

jonkoops commented on src/realm-settings/EmailTab.tsx:
```diff
@@ -0,0 +1,305 @@
+
+  const form = useForm();
+
+  useEffect(() => {
```
In case you're interested I wrote a hook that we use at Amsterdam to handle promises which allows you to pass in a promise and get the result in a structured and type-safe manner. See: https://github.com/Amsterdam/use-promise

jonkoops commented on src/realm-settings/EmailTab.tsx:
```diff
@@ -0,0 +1,305 @@
+
+    form.reset(formValues);
+
+    Object.entries(realm).map((entry) => {
+      setValue(entry[0], entry[1]);
+    });
```
You could destructure this array to make the code a bit more readable.
```suggestion
    Object.entries(realm).map(([key, value]) => {
      setValue(key, value);
    });
```

jonkoops commented on src/realm-settings/EmailTab.tsx:
```diff
@@ -0,0 +1,305 @@
+          >
+            <FormGroup label={t("from")} fieldId="kc-display-name" isRequired>
+              <TextInput
+                type="text"
```
Perhaps `type=email` would be more appropriate here?

jonkoops commented on src/realm-settings/EmailTab.tsx:
```diff
@@ -0,0 +1,305 @@
+            </FormGroup>
+            <FormGroup label={t("replyTo")} fieldId="kc-reply-to">
+              <TextInput
+                type="text"
```
Same here, this input is an e-mail address so perhaps `type=email` would be appropriate.

jenny-s51 commented:
Updated! Code looks much cleaner now; thank you for your helpful suggestions @jonkoops. I've also added email validation on the form fields with `type=email`. @ssilvert resolved conflicts as well

ssilvert commented:
@jenny-s51 Kind of odd.  The email tab no longer reads or writes any data.

jenny-s51 commented:
@ssilvert It's working for me - just fixed one of the form fields, they should all read/write data now

ssilvert commented:
@jenny-s51 sigh.  Another conflict.

jenny-s51 commented:
@ssilvert working on it now

edewit commented on src/components/scroll-form/FormPanel.tsx:
```diff
@@ -18,7 +18,7 @@ type FormPanelProps = {
 export const FormPanel = ({ title, children, scrollId }: FormPanelProps) => {
   return (
     <Card isFlat className="kc-form-panel__panel">
-      <CardHeader>
+      <CardHeader className="kc-form-panel__header">
```
do we need this class on both `Card` and `CardHeader`?

jenny-s51 replied:
In order for the styles to be applied properly each needs to have its own distinct class. I added a `className` prop to `FormPanel` so it can be specified there instead
